src/components/promos/PromoSheet.tsx

```diff
@@ -55,10 +58,30 @@ const PromoSheet = ({
   icon1,
   icon2,
   icon3,
+  campaignKey,
 }: PromoSheetProps) => {
   const { width: deviceWidth, height: deviceHeight } = useDimensions();
   const { colors } = useTheme();
   const { goBack } = useNavigation();
+  const renderedAt = Date.now();
+
+  const dismiss = useCallback(() => {
+    const timeElapsed = (Date.now() - renderedAt) / 1000;
+    analytics.track('Dismissed Feature Promo', {
+      campaign: campaignKey,
+      time_viewed: timeElapsed,
+    });
+    goBack();
+  }, [campaignKey, goBack, renderedAt]);
+
+  const engage = useCallback(() => {
+    const timeElapsed = (Date.now() - renderedAt) / 1000;
+    analytics.track('Activated Feature Promo Action', {
+      campaign: campaignKey,
+      time_viewed: timeElapsed,
+    });
+    onPress();
+  }, [campaignKey, onPress, renderedAt]);
 
```

Comment:
we should add a useEffect to trigger on sheet dismissal

something like 

`  useEffect(() => {
    return () => dismiss()
  }, []);`

Reply:
ohhh nice catch

Reply:
damn

Reply:
i wouldn't want to use `dismiss` tho right? bc that includes `goBack()`. i think i just want the analytics here